force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +137,20 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
-
+        String scientistsWithG = '';
         // Loop through the list of scientists' names
-
+        for (String scientist : scientistNames) {
             // If the name doesn't contain the letter 'G', skip this iteration
-
+            if (!scientist.toLowerCase().contains('g')) {
+                continue;
+            }
             // Add the name to the result string, followed by a comma
-
+            scientistsWithG += scientist + ',';
+        }
         // Remove the last comma from the result string
+        scientistsWithG = scientistsWithG.removeEnd(',');
 
-        return null; // Replace null with the variable you used to store the result
+        return scientistsWithG;
```

Comment:
good work. I don't think you really needed the continue here is my only change.  There is only one more line of code. If the name contains g then add it. Otherwise you will continue to the next item by default.